File: src/main/java/com/fishercoder/solutions/_1431.java
package com.fishercoder.solutions;

import java.util.ArrayList;
import java.util.List;

public class _1431 {
    public static class Solution1 {
        public List<Boolean> kidsWithCandies(int[] candies, int extraCandies) {
            int max = 0;
            for (int i : candies) {
                max = Math.max(max, i);
            }
            List<Boolean> result = new ArrayList<>();
            for (int i = 0; i < candies.length; i++) {
                if (candies[i] + extraCandies >= max) {
                    result.add(true);
                } else {
                    result.add(false);
                }
            }
            return result;
        }
    }
}

File: cpp/_1431.cpp
// Kids With the Greatest Number of Candies

class Solution {
public:
    vector<bool> kidsWithCandies(vector<int>& candies, int extraCandies) {
    
        vector<bool> v;
        int mx=0;
        
        for(int i=0;i<candies.size();i++){
            mx = max(mx,candies[i]);
        }
        
        for(int i=0;i<candies.size();i++){
            if(candies[i]+extraCandies<mx){
                v.push_back(false);
            }else{
                v.push_back(true);
            }
        }
        
        return v;
    }
};
